docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+
+### What are the benefits?
+
+Does it sound like a hype term to you? Immutabilty brings many solutions to programming matters we ecounter everyday:
+
+- Avoid side effects
+- Data changes detection made simple (shallow comparison)
+- Explicit data changes
+- Memoization
+- Memory optimization
+- Better rendering performances
+- Easy testing
+
+> « Unlike most trends in the world of JavaScript, data immutability is bound to stick with us for a while, and for good reason: firstly, because it’s not a trend: it’s a way of coding (and thinking in code) that promotes clarity, ease of use and understanding data flow, and makes code less prone to errors. »
+>
+> -- <cite>[Ricardo Magalhães](https://hackernoon.com/data-immutability-with-vanilla-javascript-63834a65a6c9)</cite>
+
+In the last few years one of our biggest challenges has been to find an efficient way to detect changes in our data to determine whether or not we should render our interfaces. It's easy to detect changes between primitive values, but it's a completely different issue for objects and arrays. You could compare them by value but you would have to implement recursive algorithms and deal with potential issues like cyclical references. Another method would be to compare object references with the strict equality operator `===`. It's more effective and there isn't any risk to enter in some deathly infinity loop. That's why modern frameworks enforce this concept.
+
+### Highlighted by modern frameworks/libraries
+
+Modern frontend frameworks and libraries are based on a concept that improves drastically performances. This is the well-known Virtual DOM. This technology has been created from a simple evidence: DOM manipulations are expensive.
+
+For the benefits explained in the previous sections, frontend frameworks and libraries chose to use immutability in order to improve their performances. Nowadays we have to deal with more and more data in our applications, and therefore more markups. So our browsers need to handle much more computations than 10 years earlier. DOM operations are expensive, modern frameworks tend to reduce the number of DOM updates.
+
+### Why do we need utilities libraries
+
+As we saw earlier the way to handle immutability in EcmaScript is made simple thanks to syntactic sugar but it finds its limits in nested structures. With the arrival of libraries like redux, nested structures have become more popular.
+
+https://gist.github.com/frinyvonnick/67664aa58d4c4f0c73e38c38d6c349a2
+
+As you can see it becomes more tedious to write and harder to read. Simple use-cases like overriding an index of an array aren't easily achievable.
+
+https://gist.github.com/frinyvonnick/0d883ce1bb07a5494318ec11c2152e92
+
+These reasons are sufficient to start finding out some tool that help focusing on what really matters. The meaning of your code.
+
+In a further part we will see how [immutadot](https://immutadot.zenika.com) allows us to address all of those problems.
```

Comment:
In a further part => In a further part, (es-tu sûr de t'engager à faire un second article ?)

Ta conclusion parle de "allows us to address all of those problems." ce qui fait un aspect assez négatif. J'indiquerai plutôt que immutadot permet de se concentrer sur la compréhension de ton code.

Je mettrai plutôt quelque chose du style:
These reasons are sufficient to start finding out some tool that help focusing on what really matters. The meaning of your code. That's why we created [immutadot](https://immutadot.zenika.com) to help us keep javascript codebase readable and maintanable. Feel free to make a try. => (demo, links etc)